### PR TITLE
Update CircleCI config

### PR DESCRIPTION
**Pull request by Omniroot**
1. Fixes CircleCI by updating the config
  - as part of this, updates to CircleCI 2.0
2. Fixes linting errors

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      #
+      # Appending an {{ epock }} to a build's cache allows rebuilds to find the most recent branch cache.
+      # For new branches, Circle falls back to matching the most recent master cache.
+      - restore_cache:
+          keys:
+            - v4-test-app-{{ .Branch }} # Match branch builds, favoring the most recent build.
+            - v4-test-app-master        # If there's no branch build match, use the most recent master cache.
+      - run: yarn run check
+      - save_cache:
+         key: v4-test-app-{{ .Branch }}-{{ epoch }}
```
Comment (lemonmade) [inline .circleci/config.yml]: I think you don’t need to use -app in all these tasks

File: package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "clean": "rimraf '*.{js,d.ts}'",
     "prebuild": "yarn run clean",
     "build": "tsc",
-    "lint": "tslint -c ./config/tslint/full.json './src/**/*.ts' --project tsconfig.json --type-check",
+    "lint": "tslint -c ./config/tslint/full.json './src/**/*.ts' --project tsconfig.json",
```
Comment (lemonmade) [inline package.json]: Why?

Comment (Omniroot) [reply package.json]:
```
--type-check is deprecated. You only need --project to enable rule which need type information.
```

File: .circleci/config.yml (hunk 1 above)
Comment (Omniroot) [reply .circleci/config.yml]: fair enough. I just copied what we had in `web`.

**Pull request merged**